File: webflux-dialogue/src/main/java/com/study/webflux/rag/domain/credit/model/source/PaymentCharge.java
package com.study.webflux.rag.domain.credit.model.source;

import com.study.webflux.rag.domain.credit.model.CreditSource;
import com.study.webflux.rag.domain.credit.model.CreditSourceType;

public record PaymentCharge(
	String paymentId,
	String pgProvider
) implements CreditSource {

	@Override
	public CreditSourceType sourceType() {
		return CreditSourceType.PAYMENT_CHARGE;
	}
}

File: backend/domain/src/main/kotlin/com/miyou/app/domain/credit/model/PaymentCharge.kt
package com.miyou.app.domain.credit.model

data class PaymentCharge(
    val paymentId: String,
    val pgProvider: String,
) : CreditSource {
    override fun sourceType(): CreditSourceType = CreditSourceType.PAYMENT_CHARGE

    fun paymentId(): String = paymentId

    fun pgProvider(): String = pgProvider
}
